reward/reward_distributor_test.go

```diff
@@ -83,35 +83,68 @@ func Test_isEmptyAddress(t *testing.T) {
 func TestRewardDistributor_getTotalTxFee(t *testing.T) {
 	testCases := []struct {
 		gasUsed            uint64
-		unitPrice          uint64
+		baseFee            *big.Int
 		expectedTotalTxFee *big.Int
 	}{
-		{0, 25000000000, big.NewInt(0)},
-		{200000, 25000000000, big.NewInt(5000000000000000)},
-		{129346, 10000000000, big.NewInt(1293460000000000)},
-		{9236192, 50000, big.NewInt(461809600000)},
-		{12936418927364923, 0, big.NewInt(0)},
+		{0, big.NewInt(25000000000), big.NewInt(0)},
+		{200000, big.NewInt(25000000000), big.NewInt(5000000000000000)},
+		{200000, big.NewInt(25000000000), big.NewInt(5000000000000000)},
+		{129346, big.NewInt(10000000000), big.NewInt(1293460000000000)},
+		{129346, big.NewInt(10000000000), big.NewInt(1293460000000000)},
+		{9236192, big.NewInt(50000), big.NewInt(461809600000)},
+		{9236192, big.NewInt(50000), big.NewInt(461809600000)},
+		{12936418927364923, big.NewInt(0), big.NewInt(0)},
 	}
 	rewardDistributor := NewRewardDistributor(newDefaultTestGovernance())
 	rewardConfig := &rewardConfig{}
 
 	header := &types.Header{}
-	unitPrice := big.NewInt(0)
 
 	for _, testCase := range testCases {
 		header.GasUsed = testCase.gasUsed
-		rewardConfig.unitPrice = unitPrice.SetUint64(testCase.unitPrice)
+		header.BaseFee = testCase.baseFee
```

Comment:
I think before/after fork test should be separated such as below. Also, after fork, unit price should not matter.

```go
func TestRewardDistributor_getTotalTxFee(t *testing.T) {
	testCases := []struct {
		gasUsed            uint64
		unitPrice          uint64
		baseFee            *big.Int
		expectedTotalTxFee *big.Int
	}{
		// before fork
		{0, 25000000000, nil, big.NewInt(0)},
		{200000, 25000000000, nil, big.NewInt(5000000000000000)},
		{129346, 10000000000, nil, big.NewInt(1293460000000000)},
		{9236192, 50000, nil, big.NewInt(461809600000)},
		{12936418927364923, 0, nil, big.NewInt(0)},

		// after fork
		{0, 0, big.NewInt(25000000000), big.NewInt(0)},
		{200000, 0, big.NewInt(25000000000), big.NewInt(5000000000000000)},
		{129346, 0, big.NewInt(10000000000), big.NewInt(1293460000000000)},

		// after fork, arbitrary unitprice
		{9236192, 111111, big.NewInt(50000), big.NewInt(461809600000)},
		{12936418927364923, 222222, big.NewInt(0), big.NewInt(0)},
	}
	rewardDistributor := NewRewardDistributor(newDefaultTestGovernance())
	rewardConfig := &rewardConfig{}

	header := &types.Header{}
	unitPrice := big.NewInt(0)

	for _, testCase := range testCases {
		header.GasUsed = testCase.gasUsed
		header.BaseFee = testCase.baseFee
		rewardConfig.unitPrice = unitPrice.SetUint64(testCase.unitPrice)
```

Reply:
Looks more clear. I added it. Thanks